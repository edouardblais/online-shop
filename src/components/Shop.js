import React from "react";
import Data from './Data';
import { Link } from "react-router-dom";

const Shop = ({addToCart, 
               addAcres, 
               message,
               numberWithSep,
            }) => {
    return (
        <div className="bg-black flex flex-col items-center flex-grow min-h-screen p-24 font-mono">
            <div className="bg-black text-white flex justify-center flex-col grow gap-4 p-4 max-w-[65%] min-w-[25%] rounded">
                <div className="flex flex-col gap-6 flex-wrap justify-center items-center">
                    {Data?.map((mooninfo) => {
                        return ( <div key={mooninfo.name} className="flex flex-row gap-4">
                                    <div>
                                        <img alt={mooninfo.name} src={mooninfo.image} className="aspect-auto max-w-lg"></img>
                                    </div>
                                    <div className="flex flex-col gap-4 justify-center p-2">
                                        <div className="flex flex-row">
                                            <div className="flex flex-col">
                                                <div className="font-bold text-xl">{mooninfo.name}</div>
                                                <div className="italic">Moon of {mooninfo.planet}</div>
                                            </div>
                                            <div className="grow"></div>
                                            <div className="text-md">$ {numberWithSep(mooninfo.price)} per acre</div>
                                        </div>
                                        <div className="flex flex-col">
                                            <form method="post" action="" className="flex flex-row gap-2">
                                                <label htmlFor={mooninfo.name} className="text-md">Number of acres desired on {mooninfo.name}</label>
                                                <div className="grow"></div>
                                                <input className='text-black max-w-[22%] rounded' type='number' id={mooninfo.name} name={mooninfo.name} onChange={(e) => addAcres(e, mooninfo)}></input>
                                            </form>
                                            <div className="font-bold text-md">{message}</div>
                                        </div>
                                        <div className="flex flex-row items-center">
                                            <div className="flex items-center justify-center">
                                                <Link to={mooninfo.name} className="text-xl rounded bg-black text-white p-1 border-white border-2">See Info</Link >
                                            </div>
                                            <div className="grow"></div>
                                            <div className="flex items-center justify-center">
                                                <button className="text-xl rounded bg-black text-white p-1 border-white border-2" type='button' onClick={() => addToCart(mooninfo)}>Add to Cart</button>
                                            </div>
                                        </div>
                                    </div>
                                </div> 
                        )
                    })}
                </div>
            </div>
        </div>
    )
}

export default Shop