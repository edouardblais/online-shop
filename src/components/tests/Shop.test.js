import { MemoryRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer'
import Shop from '../Shop';
import { render, screen }from '@testing-library/react'
import userEvent from '@testing-library/user-event';

describe('Shop component', () => {
    it('renders correctly', () => {
        const shop = TestRenderer
            .create(
                <MemoryRouter>
                    <Shop />)
                </MemoryRouter>
            )
            .toJSON();
        expect(shop).toMatchSnapshot();
    })

    it('calls onChange correct amount of times', () => {
        const addAcresMock = jest.fn();
        render(<Shop addAcres={addAcresMock} />, {wrapper: MemoryRouter});
        const input = screen.getByRole('textbox', { name: 'Number of acres desired on Callisto:'});
        
        userEvent.type(input, '100');

        expect(addAcresMock).toHaveBeenCalledTimes(3);
    })

    it('renders appropriate image', () => {
        render(<Shop/>, {wrapper: MemoryRouter});
        const img = screen.getByAltText('Io');

        expect(img.src).toBe('http://localhost/io.jpg')
    })
})
