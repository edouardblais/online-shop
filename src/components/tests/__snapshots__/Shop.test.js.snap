// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop component renders correctly 1`] = `
Array [
  <div
    className="bg-black flex flex-col items-center flex-grow min-h-screen p-24"
  >
    <div
      className="bg-black text-white flex justify-center flex-col grow gap-4 p-4 max-w-[60%] min-w-[25%] rounded"
    >
      <div
        className="flex flex-col gap-6 flex-wrap justify-center items-center"
      >
        <div
          className="flex flex-row gap-4"
        >
          <div>
            <img
              alt="The Moon"
              className="aspect-auto max-w-lg"
              src="moon.jpg"
            />
          </div>
          <div
            className="flex flex-col gap-4 justify-center p-2"
          >
            <div
              className="flex flex-row"
            >
              <div
                className="font-bold text-xl"
              >
                The Moon
              </div>
              <div
                className="grow"
              />
              <div
                className="text-md"
              >
                $ 
                 per acre
              </div>
            </div>
            <div
              className="flex flex-col"
            >
              <form
                action=""
                className="flex flex-row gap-2"
                method="post"
              >
                <label
                  className="text-md"
                  htmlFor="The Moon"
                >
                  Number of acres desired on 
                  The Moon
                </label>
                <div
                  className="grow"
                />
                <input
                  className="text-black max-w-[22%] rounded"
                  id="The Moon"
                  name="The Moon"
                  onChange={[Function]}
                  type="number"
                />
              </form>
              <div
                className="font-bold text-md"
              />
            </div>
            <div
              className="flex flex-row items-center"
            >
              <div
                className="flex items-center justify-center"
              >
                <a
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  href="/The Moon"
                  onClick={[Function]}
                >
                  See Info
                </a>
              </div>
              <div
                className="grow"
              />
              <div
                className="flex items-center justify-center"
              >
                <button
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  onClick={[Function]}
                  type="button"
                >
                  Add to Cart
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="flex flex-row gap-4"
        >
          <div>
            <img
              alt="Callisto"
              className="aspect-auto max-w-lg"
              src="Callisto.jpg"
            />
          </div>
          <div
            className="flex flex-col gap-4 justify-center p-2"
          >
            <div
              className="flex flex-row"
            >
              <div
                className="font-bold text-xl"
              >
                Callisto
              </div>
              <div
                className="grow"
              />
              <div
                className="text-md"
              >
                $ 
                 per acre
              </div>
            </div>
            <div
              className="flex flex-col"
            >
              <form
                action=""
                className="flex flex-row gap-2"
                method="post"
              >
                <label
                  className="text-md"
                  htmlFor="Callisto"
                >
                  Number of acres desired on 
                  Callisto
                </label>
                <div
                  className="grow"
                />
                <input
                  className="text-black max-w-[22%] rounded"
                  id="Callisto"
                  name="Callisto"
                  onChange={[Function]}
                  type="number"
                />
              </form>
              <div
                className="font-bold text-md"
              />
            </div>
            <div
              className="flex flex-row items-center"
            >
              <div
                className="flex items-center justify-center"
              >
                <a
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  href="/Callisto"
                  onClick={[Function]}
                >
                  See Info
                </a>
              </div>
              <div
                className="grow"
              />
              <div
                className="flex items-center justify-center"
              >
                <button
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  onClick={[Function]}
                  type="button"
                >
                  Add to Cart
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="flex flex-row gap-4"
        >
          <div>
            <img
              alt="Io"
              className="aspect-auto max-w-lg"
              src="io.jpg"
            />
          </div>
          <div
            className="flex flex-col gap-4 justify-center p-2"
          >
            <div
              className="flex flex-row"
            >
              <div
                className="font-bold text-xl"
              >
                Io
              </div>
              <div
                className="grow"
              />
              <div
                className="text-md"
              >
                $ 
                 per acre
              </div>
            </div>
            <div
              className="flex flex-col"
            >
              <form
                action=""
                className="flex flex-row gap-2"
                method="post"
              >
                <label
                  className="text-md"
                  htmlFor="Io"
                >
                  Number of acres desired on 
                  Io
                </label>
                <div
                  className="grow"
                />
                <input
                  className="text-black max-w-[22%] rounded"
                  id="Io"
                  name="Io"
                  onChange={[Function]}
                  type="number"
                />
              </form>
              <div
                className="font-bold text-md"
              />
            </div>
            <div
              className="flex flex-row items-center"
            >
              <div
                className="flex items-center justify-center"
              >
                <a
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  href="/Io"
                  onClick={[Function]}
                >
                  See Info
                </a>
              </div>
              <div
                className="grow"
              />
              <div
                className="flex items-center justify-center"
              >
                <button
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  onClick={[Function]}
                  type="button"
                >
                  Add to Cart
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="flex flex-row gap-4"
        >
          <div>
            <img
              alt="Charon"
              className="aspect-auto max-w-lg"
              src="charon.jpg"
            />
          </div>
          <div
            className="flex flex-col gap-4 justify-center p-2"
          >
            <div
              className="flex flex-row"
            >
              <div
                className="font-bold text-xl"
              >
                Charon
              </div>
              <div
                className="grow"
              />
              <div
                className="text-md"
              >
                $ 
                 per acre
              </div>
            </div>
            <div
              className="flex flex-col"
            >
              <form
                action=""
                className="flex flex-row gap-2"
                method="post"
              >
                <label
                  className="text-md"
                  htmlFor="Charon"
                >
                  Number of acres desired on 
                  Charon
                </label>
                <div
                  className="grow"
                />
                <input
                  className="text-black max-w-[22%] rounded"
                  id="Charon"
                  name="Charon"
                  onChange={[Function]}
                  type="number"
                />
              </form>
              <div
                className="font-bold text-md"
              />
            </div>
            <div
              className="flex flex-row items-center"
            >
              <div
                className="flex items-center justify-center"
              >
                <a
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  href="/Charon"
                  onClick={[Function]}
                >
                  See Info
                </a>
              </div>
              <div
                className="grow"
              />
              <div
                className="flex items-center justify-center"
              >
                <button
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  onClick={[Function]}
                  type="button"
                >
                  Add to Cart
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="flex flex-row gap-4"
        >
          <div>
            <img
              alt="Enceladus"
              className="aspect-auto max-w-lg"
              src="enceladus.jpg"
            />
          </div>
          <div
            className="flex flex-col gap-4 justify-center p-2"
          >
            <div
              className="flex flex-row"
            >
              <div
                className="font-bold text-xl"
              >
                Enceladus
              </div>
              <div
                className="grow"
              />
              <div
                className="text-md"
              >
                $ 
                 per acre
              </div>
            </div>
            <div
              className="flex flex-col"
            >
              <form
                action=""
                className="flex flex-row gap-2"
                method="post"
              >
                <label
                  className="text-md"
                  htmlFor="Enceladus"
                >
                  Number of acres desired on 
                  Enceladus
                </label>
                <div
                  className="grow"
                />
                <input
                  className="text-black max-w-[22%] rounded"
                  id="Enceladus"
                  name="Enceladus"
                  onChange={[Function]}
                  type="number"
                />
              </form>
              <div
                className="font-bold text-md"
              />
            </div>
            <div
              className="flex flex-row items-center"
            >
              <div
                className="flex items-center justify-center"
              >
                <a
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  href="/Enceladus"
                  onClick={[Function]}
                >
                  See Info
                </a>
              </div>
              <div
                className="grow"
              />
              <div
                className="flex items-center justify-center"
              >
                <button
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  onClick={[Function]}
                  type="button"
                >
                  Add to Cart
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="flex flex-row gap-4"
        >
          <div>
            <img
              alt="Ganymede"
              className="aspect-auto max-w-lg"
              src="ganymede.jpg"
            />
          </div>
          <div
            className="flex flex-col gap-4 justify-center p-2"
          >
            <div
              className="flex flex-row"
            >
              <div
                className="font-bold text-xl"
              >
                Ganymede
              </div>
              <div
                className="grow"
              />
              <div
                className="text-md"
              >
                $ 
                 per acre
              </div>
            </div>
            <div
              className="flex flex-col"
            >
              <form
                action=""
                className="flex flex-row gap-2"
                method="post"
              >
                <label
                  className="text-md"
                  htmlFor="Ganymede"
                >
                  Number of acres desired on 
                  Ganymede
                </label>
                <div
                  className="grow"
                />
                <input
                  className="text-black max-w-[22%] rounded"
                  id="Ganymede"
                  name="Ganymede"
                  onChange={[Function]}
                  type="number"
                />
              </form>
              <div
                className="font-bold text-md"
              />
            </div>
            <div
              className="flex flex-row items-center"
            >
              <div
                className="flex items-center justify-center"
              >
                <a
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  href="/Ganymede"
                  onClick={[Function]}
                >
                  See Info
                </a>
              </div>
              <div
                className="grow"
              />
              <div
                className="flex items-center justify-center"
              >
                <button
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  onClick={[Function]}
                  type="button"
                >
                  Add to Cart
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="flex flex-row gap-4"
        >
          <div>
            <img
              alt="Europa"
              className="aspect-auto max-w-lg"
              src="europa.jpg"
            />
          </div>
          <div
            className="flex flex-col gap-4 justify-center p-2"
          >
            <div
              className="flex flex-row"
            >
              <div
                className="font-bold text-xl"
              >
                Europa
              </div>
              <div
                className="grow"
              />
              <div
                className="text-md"
              >
                $ 
                 per acre
              </div>
            </div>
            <div
              className="flex flex-col"
            >
              <form
                action=""
                className="flex flex-row gap-2"
                method="post"
              >
                <label
                  className="text-md"
                  htmlFor="Europa"
                >
                  Number of acres desired on 
                  Europa
                </label>
                <div
                  className="grow"
                />
                <input
                  className="text-black max-w-[22%] rounded"
                  id="Europa"
                  name="Europa"
                  onChange={[Function]}
                  type="number"
                />
              </form>
              <div
                className="font-bold text-md"
              />
            </div>
            <div
              className="flex flex-row items-center"
            >
              <div
                className="flex items-center justify-center"
              >
                <a
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  href="/Europa"
                  onClick={[Function]}
                >
                  See Info
                </a>
              </div>
              <div
                className="grow"
              />
              <div
                className="flex items-center justify-center"
              >
                <button
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  onClick={[Function]}
                  type="button"
                >
                  Add to Cart
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="flex flex-row gap-4"
        >
          <div>
            <img
              alt="Phobos"
              className="aspect-auto max-w-lg"
              src="phobos.jpg"
            />
          </div>
          <div
            className="flex flex-col gap-4 justify-center p-2"
          >
            <div
              className="flex flex-row"
            >
              <div
                className="font-bold text-xl"
              >
                Phobos
              </div>
              <div
                className="grow"
              />
              <div
                className="text-md"
              >
                $ 
                 per acre
              </div>
            </div>
            <div
              className="flex flex-col"
            >
              <form
                action=""
                className="flex flex-row gap-2"
                method="post"
              >
                <label
                  className="text-md"
                  htmlFor="Phobos"
                >
                  Number of acres desired on 
                  Phobos
                </label>
                <div
                  className="grow"
                />
                <input
                  className="text-black max-w-[22%] rounded"
                  id="Phobos"
                  name="Phobos"
                  onChange={[Function]}
                  type="number"
                />
              </form>
              <div
                className="font-bold text-md"
              />
            </div>
            <div
              className="flex flex-row items-center"
            >
              <div
                className="flex items-center justify-center"
              >
                <a
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  href="/Phobos"
                  onClick={[Function]}
                >
                  See Info
                </a>
              </div>
              <div
                className="grow"
              />
              <div
                className="flex items-center justify-center"
              >
                <button
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  onClick={[Function]}
                  type="button"
                >
                  Add to Cart
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="flex flex-row gap-4"
        >
          <div>
            <img
              alt="Titan"
              className="aspect-auto max-w-lg"
              src="titan.jpg"
            />
          </div>
          <div
            className="flex flex-col gap-4 justify-center p-2"
          >
            <div
              className="flex flex-row"
            >
              <div
                className="font-bold text-xl"
              >
                Titan
              </div>
              <div
                className="grow"
              />
              <div
                className="text-md"
              >
                $ 
                 per acre
              </div>
            </div>
            <div
              className="flex flex-col"
            >
              <form
                action=""
                className="flex flex-row gap-2"
                method="post"
              >
                <label
                  className="text-md"
                  htmlFor="Titan"
                >
                  Number of acres desired on 
                  Titan
                </label>
                <div
                  className="grow"
                />
                <input
                  className="text-black max-w-[22%] rounded"
                  id="Titan"
                  name="Titan"
                  onChange={[Function]}
                  type="number"
                />
              </form>
              <div
                className="font-bold text-md"
              />
            </div>
            <div
              className="flex flex-row items-center"
            >
              <div
                className="flex items-center justify-center"
              >
                <a
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  href="/Titan"
                  onClick={[Function]}
                >
                  See Info
                </a>
              </div>
              <div
                className="grow"
              />
              <div
                className="flex items-center justify-center"
              >
                <button
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  onClick={[Function]}
                  type="button"
                >
                  Add to Cart
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="flex flex-row gap-4"
        >
          <div>
            <img
              alt="Triton"
              className="aspect-auto max-w-lg"
              src="triton.jpg"
            />
          </div>
          <div
            className="flex flex-col gap-4 justify-center p-2"
          >
            <div
              className="flex flex-row"
            >
              <div
                className="font-bold text-xl"
              >
                Triton
              </div>
              <div
                className="grow"
              />
              <div
                className="text-md"
              >
                $ 
                 per acre
              </div>
            </div>
            <div
              className="flex flex-col"
            >
              <form
                action=""
                className="flex flex-row gap-2"
                method="post"
              >
                <label
                  className="text-md"
                  htmlFor="Triton"
                >
                  Number of acres desired on 
                  Triton
                </label>
                <div
                  className="grow"
                />
                <input
                  className="text-black max-w-[22%] rounded"
                  id="Triton"
                  name="Triton"
                  onChange={[Function]}
                  type="number"
                />
              </form>
              <div
                className="font-bold text-md"
              />
            </div>
            <div
              className="flex flex-row items-center"
            >
              <div
                className="flex items-center justify-center"
              >
                <a
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  href="/Triton"
                  onClick={[Function]}
                >
                  See Info
                </a>
              </div>
              <div
                className="grow"
              />
              <div
                className="flex items-center justify-center"
              >
                <button
                  className="font-bold text-lg rounded bg-black text-white p-1 border-white border-2"
                  onClick={[Function]}
                  type="button"
                >
                  Add to Cart
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  ")",
]
`;
